### PR TITLE
feat: add --fast-dev option for faster development builds

### PR DESCRIPTION
**Pull request by ZauberNerd**
<details>
<summary>Bors merge bot cheat sheet</summary>

We are using [bors-ng](https://github.com/bors-ng/bors-ng) to automate merging of our pull requests. The following table provides a summary of commands that are available to reviewers (members of this repository with push access) and delegates (in case of `bors delegate+` or `bors delegate=[list]`).

| Syntax | Description |
| --- | --- |
| bors merge | Run the test suite and push to master if it passes. Short for "reviewed: looks good." |
| bors merge- | Cancel an r+, r=, merge, or merge= |
| bors try | Run the test suite without pushing to master. |
| bors try- | Cancel a try |
| bors delegate+ | Allow the pull request author to merge their changes. |
| bors delegate=[list] | Allow the listed users to r+ this pull request's changes. |
| bors retry | Run the previous command a second time. |

This is a short collection of opinionated commands. For a full list of the commands read the [bors reference](https://bors.tech/documentation/).

</details>

### CODE REVIEW

File: packages/webpack/mixins/fast-dev/mixin.core.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      target === 'build' ||
+      (target === 'node' && process.env.NODE_ENV === 'production')
+    ) {
+      return;
+    }
+
+    if (this.options.fastDev) {
+      process.env.__HOPS_FAST_DEV__ = 'true';
+
+      jsLoaderConfig.exclude = [/node_modules\/(?!(@xingternal|hops-.*)\/).*/];
```
Comment (herschel666) [inline packages/webpack/mixins/fast-dev/mixin.core.js:23]: `@xingternal`? what is this? never heard of it! 😜

**Comment by KnisterPeter** [comment]
We should add more to this. The idea was to create an option to improve dx by skipping all optimizations possible for modern browsers.

File: packages/webpack/mixins/fast-dev/mixin.core.js (hunk 1 above)
Comment (ZauberNerd) [reply packages/webpack/mixins/fast-dev/mixin.core.js:23]: Oh, you're right. This is breaking the separation and it should be handled in our internal mixins.

**Comment by ZauberNerd** [comment]
bors merge

**Pull request merged**